base/compiler/abstractinterpretation.jl

```diff
@@ -297,18 +307,16 @@ function abstract_call_method(method::Method, @nospecialize(sig), sparams::Simpl
         recomputed = ccall(:jl_type_intersection_with_env, Any, (Any, Any), sig, method.sig)::SimpleVector
         sig = recomputed[1]
         if !isa(unwrap_unionall(sig), DataType) # probably Union{}
-            return Any, false
+            return Any, false, nothing
         end
         sparams = recomputed[2]::SimpleVector
     end
 
     rt, edge = typeinf_edge(method, sig, sparams, sv)
     if edge === nothing
         edgecycle = true
-    else
-        add_backedge!(edge::MethodInstance, sv)
```

Comment:
if `edgecycle` is true, this is the wrong edge to create, but it also means that we will compute and add the real edge later

edit: theoretically, we should be able to see the `Any` information result and use that to break the callgraph cycle on this edge here (possibly permitting faster convergence), but that does weird things to the convergence algorithm postulates